udata_search_service/presentation/api.py

```diff
@@ -175,10 +283,89 @@ def reuses_search(reuse_service: ReuseService = Provide[Container.reuse_service]
                          request_args.page_size, next_url, prev_url)
 
 
-@bp.route("/reuses/<reuse_id>/", methods=["GET"])
+@bp.route("/reuses/<reuse_id>/", methods=["GET"], endpoint='reuse_get_specific')
 @inject
 def get_reuse(reuse_id: str, reuse_service: ReuseService = Provide[Container.reuse_service]):
     result = reuse_service.find_one(reuse_id)
     if result:
         return jsonify(result)
     abort(404, 'reuse not found')
+
+
+@bp.route("/reuses/index", methods=["POST"], endpoint='reuse_index')
+@inject
+def reuse_index(reuse_service: ReuseService = Provide[Container.reuse_service]):
+    try:
+        validated_obj = ReuseToIndex(**request.json['data'])
+    except ValidationError as e:
+        abort(400, e)
+
+    document = ReuseConsumer.load_from_dict(validated_obj.dict())
+    reuse_service.feed(document)
+    return jsonify({'data': 'Reuse added to index'})
+
+
+@bp.route("/reuses/<reuse_id>/unindex", methods=["DELETE"], endpoint='reuse_unindex')
+@inject
+def reuse_unindex(reuse_id: str, reuse_service: ReuseService = Provide[Container.reuse_service]):
+    reuse_service.delete_one(reuse_id)
+    return jsonify({'data': 'Reuse removed from index'})
+
+
+class RequestReindex(BaseModel):
+    key_id: str
+    document: Optional[dict] = {}
+    message_type: str
+    index: str
+
+
+@bp.route("/reindex", methods=["POST"], endpoint='reindex')
+@inject
+def general_reindex(search_client: ElasticClient = Provide[Container.search_client]):
+    try:
+        try:
+            validated_obj = RequestReindex(**request.json)
+        except ValidationError as e:
+            abort(400, e)
+
+        message_type, index_name, data = parse_message(validated_obj.dict())
+        index_name = f'{Config.UDATA_INSTANCE_NAME}-{index_name}'
+        if message_type == EventMessageType.REINDEX.value:
```

Comment:
I think we can clean a bit of the logic here? The endpoint `/reindex` doesn't need a "message type" for example. Could we also do without `key_id`, as we do in other index functions
Could we have a body closer to `/index` endpoints? Putting the index creation in a dedicated function, and having the rest being similar